lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -996,6 +1019,29 @@ def exec_module(self, **kwargs):
                     differences.append('Zones')
                     changed = True
 
+                if 'diagnosticsProfile' not in vm_dict['properties']:
+                    vm_dict['properties']['diagnosticsProfile'] = {}
+                if 'bootDiagnostics' not in vm_dict['properties']['diagnosticsProfile']:
```

Comment:
elif? or vm_dict['properties'].get('diagnosticsProfile')

Reply:
I cannot guarantee that Azure will always return a diagnosticsProfile.  In fact, I don't think it does when the feature has never been configured.  So the first "if" statement is correct.

I cannot guarantee that Azure will always return a bootDiagnostics inside the diagnosticsProfile, therefore the second "if" is also correct.  elif would cause it to do the wrong thing